basedtyping/__init__.py

```diff
@@ -23,15 +25,22 @@
     _remove_dups_flatten,
     _SpecialForm,
     _tp_cache,
-    _type_check,
     cast,
 )
 
 import typing_extensions
-from typing_extensions import Never, ParamSpec, Self, TypeAlias, TypeGuard, TypeVarTuple
+from typing_extensions import Never, ParamSpec, Self, TypeAlias, TypeGuard, TypeVarTuple, override
 
+from basedtyping import transformer
 from basedtyping.runtime_only import OldUnionType
 
+BASEDMYPY_TYPE_CHECKING: Final = False
+"""a special constant, is always `False`, but basedmypy will always assume it to be true
+
+if you aren't using basedmypy, you may have to configure your type checker to consider
+ this variable "always false"
```

Comment:
why would you need to do that? any other type checker would treat it as `Literal[False]` unless it was special cased or configured to behave otherwise

Reply:
there are like 6 different kinds of `False`, it's a mess

Reply:
ah, i'll need to make some change in basedmypy for atht

Reply:
ergh, to be discussed

Reply:
i put an issue and a comment

Reply:
i still dont get it, that issue seems to be unrelated to this docstring. if you arent using basedmypy, how would you run into this issue?

Reply:
https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoBiqAhgDYA0UIApgG7VkD68C1AsAFCecBCAggGUAogBEAsgE0AChMYAVaUMYBhABJDlAaQCSAOQDiALkIlSUALyEyAZ3Zd7SYFH7Dx02QqlK1GnQcOcUEFQxBaYKDAAFACUnNSktgEcwSFh1jAgMdwcNPRMLNSRxNFBAMTh%2BAA%2BUOkg2UA

we need to fix it in bmp then we can update it to `Literal[False]`